### PR TITLE
Executable

### PR DESCRIPTION
**Pull request by psunnn**
Paving the way for templates.

### CODE REVIEW

File: clients/cluster/cluster.go [hunk 1]
```diff
@@ -17,7 +17,7 @@ import (
 type Client interface {
 	Name() string
 	Initialize(conf config.Config) error
-	CanBeRun(clusterName string, definition state.Definition) (bool, error)
+	CanBeRun(clusterName string, executableResources state.ExecutableResources) (bool, error)
```
Comment (ujjwalsfix) [inline clients/cluster/cluster.go]: look into `flotilla-sf` for this

File: clients/cluster/ecs_cluster_client_test.go [hunk 2]
```diff
@@ -161,10 +161,10 @@ func TestECSClusterClient_CanBeRun(t *testing.T) {
 	tooMuch := int64(100)
 	justRight := int64(99)
 
-	unrunnable := state.Definition{
+	unrunnable := state.ExecutableResources{
 		Memory: &tooMuch,
```
Comment (ujjwalsfix) [inline clients/cluster/ecs_cluster_client_test.go]: cute naming :) `tooMuch`

File: flotilla/endpoints.go [hunk 3]
```diff
@@ -344,7 +356,21 @@ func (ep *endpoints) CreateRunV2(w http.ResponseWriter, r *http.Request) {
 	} else {
 		lr.Engine = &state.DefaultEngine
 	}
-	run, err := ep.executionService.Create(vars["definition_id"], lr.ClusterName, lr.Env, lr.RunTags.OwnerEmail, nil, nil, nil, lr.Engine, nil, nil)
+
+	req := state.DefinitionExecutionRequest{
+		ExecutionRequestCommon: state.ExecutionRequestCommon{
```
Comment (ujjwalsfix) [inline flotilla/endpoints.go]: Maybe an opportunity to make this more DRY, we have the same block repeated a few times.

File: clients/cluster/cluster.go (hunk 1 above)
Comment (psunnn) [reply clients/cluster/cluster.go]: 👍

File: clients/cluster/ecs_cluster_client_test.go (hunk 2 above)
Comment (psunnn) [reply clients/cluster/ecs_cluster_client_test.go]: shout out @thedatachef 😂

**Pull request merged**